### PR TITLE
Add time&mem test for snake oil plotting

### PR DESCRIPTION
**Pull request by yngve-sk**
Makes it easier to quantify regressions/improvements in memory usage/performance for plotting
Approach: Does PlotAPI calls equivalent of going through all response keys in the plotter, on the snake oil case
Running memory takes ~20s, testing time takes <5s (on py312 MBP max)

### CODE REVIEW

File: tests/ert/unit_tests/gui/tools/plot/test_plot_api.py [hunk 1]
```diff
@@ -275,6 +293,75 @@ def test_plot_api_big_summary_memory_usage(
...
+                    key_info.key,
+                    [e.id for e in all_ensembles],
+                )
+
+    t2 = time.time()
+    time_to_get_metadata = t1 - t0
+    time_to_cycle_through_responses = t2 - t1
+
+    assert time_to_get_metadata < 1  # 0.09 on py312 macbook pro m1 max
+    assert time_to_cycle_through_responses < 14  # 4.48 on py312 macbook pro m1 max
```
Comment (oyvindeide) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_api.py:326]: Suggest removing the comments and just add a single saying something like:
```python
# Local times were about 10% of the asserted times

File: tests/ert/unit_tests/gui/tools/plot/test_plot_api.py [hunk 2]
```diff
@@ -275,6 +293,75 @@ def test_plot_api_big_summary_memory_usage(
...
+
+    gc.collect()
```
Comment (oyvindeide) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_api.py:328]: Do you need this? You already call it in the fixture

File: tests/ert/unit_tests/gui/tools/plot/test_plot_api.py [hunk 3]
```diff
@@ -275,6 +293,75 @@ def test_plot_api_big_summary_memory_usage(
...
+                        [e.id for e in all_ensembles],
+                    )
+
+    stats = memray._memray.compute_statistics("memray.bin")
+    os.remove("memray.bin")
+    total_memory_mb = stats.total_memory_allocated / (1024**2)
+    peak_memory_mb = stats.peak_memory_allocated / (1024**2)
+
+    assert total_memory_mb < 5000  # Tested locally to 3579mb on macbook pro m1 max
+    assert peak_memory_mb < 1500  # Tested locally to 840mb on macbook pro m1 max
```
Comment (oyvindeide) [inline tests/ert/unit_tests/gui/tools/plot/test_plot_api.py:362]: Same comment as with time

**Review by oyvindeide — COMMENTED** [review]
Looks good, nice to have some direct tests on PlotAPI 👍 Did you check the potential flakyness of these tests?

**Comment by yngve-sk** [comment]
> Looks good, nice to have some direct tests on PlotAPI 👍 Did you check the potential flakyness of these tests?

I did run it 25 times with pytest-repeat locally and it did not fail. Did not find them to be flaky so far, did not fail on PR runners either with 2 reruns (now running once more just to check).

File: tests/ert/unit_tests/gui/tools/plot/test_plot_api.py (hunk 2 above)
Comment (yngve-sk) [reply tests/ert/unit_tests/gui/tools/plot/test_plot_api.py]: Oops, removed it!

**Review by oyvindeide — APPROVED** [review]
LGTM! Remember to squash commits

**Pull request merged**